File: Stack/0020-valid-parentheses/0020-valid-parentheses.py

from collections import deque
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        mapping = {"(" : ")", "{" : "}", "[" : "]"}

       
        for c in range(0, len(s)):
            if s[c] in "({[":
                stack.append(s[c])
            elif len(stack)==0 or s[c] != mapping[stack.pop()]:
                return False
        return len(stack) == 0 


        #     if s[c] == "[" or s[c] == "(" or s[c] == "{":
        #         stack.append(s[c])

        # if len(stack) == 0:
        #     return False    

        # for c in range(0, len(s)):
        #     if s[c] == "}":
        #         if stack[-1] == "{":
        #             stack.pop()
        #         else:
        #             return False
        #     if s[c] == "]":
        #         if stack[-1] == "[":
        #             stack.pop()
        #         else:
        #             return False
        #     if s[c] == ")":
        #         if stack[-1] == "(":
        #             stack.pop()
        #         else:
        #             return False
        # return True
        

File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<String> parenths = new Stack<>();
        String[] split = s.split("");
        System.out.print(split[0]);
        for(int i = 0; i < split.length; i++){
            if(split[i].equals("{") || split[i].equals("[") || split[i].equals("(")){
                parenths.push(split[i]);
            }
            else if(parenths.empty() || 
                (split[i].equals("}") && !parenths.pop().equals("{")) ||
                (split[i].equals(")") && !parenths.pop().equals("("))|| 
                (split[i].equals("]") && !parenths.pop().equals("["))){
                return false;
            }
        }
        return parenths.empty();
    }
}